src/frontend/src/eth/services/send.services.ts

```diff
@@ -508,14 +511,45 @@ const approve = async ({
 			erc20HelperContractAddress
 		})
 	) {
-		return { transactionApproved: false };
+		return { transactionApproved: false, nonce };
 	}
 
+	const preApprovedAmount = await erc20ContractAllowance({
+		token,
+		owner: from,
+		spender: erc20HelperContractAddress,
+		networkId
+	});
+
+	// If there is already an approved allowance that is enough for the required amount, we don't need to approve again.
+	if (preApprovedAmount.gte(amount)) {
+		progress(ProgressStepsSend.APPROVE);
+		return { transactionApproved: false, nonce };
+	}
+
+	// If the existing pre-approved amount is not enough but non-null, we need to reset the allowance first and then approve the new amount.
+	if (preApprovedAmount.gt(ZERO)) {
+		await prepareAndSignApproval({
+			...rest,
+			nonce,
+			amount: ZERO,
+			token,
+			sourceNetwork,
+			spender: erc20HelperContractAddress
+		});
+	}
+
+	const nonceApproval = preApprovedAmount.gt(ZERO) ? nonce + 1 : nonce;
```

Comment:
Can you add a comment about why we increment the nonce here? I asked myself why we do so because the caller of the function also increment the nonce (`nonceTransaction`).

Reply:
done